### PR TITLE
[JENKINS-36301] Update to the new Parent POM

### PR DESCRIPTION
**Pull request by kwhetstone**
[JENKINS-36301](https://issues.jenkins-ci.org/browse/JENKINS-36301)

Convert to use parent pom.  Since the tests used old features of HtmlUnit, I took this opportunity to convert these tests to Junit4.  Also kept this at jenkins.version=1.544.3 since there didn't seem to be any issues with this version.  That can be bumped if needed.  Tests against it's current version and the 2 line pass.

@reviewbybees

### CODE REVIEW

**Comment by ghost** [comment]
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

File: pom.xml [hunk 1]
```diff
@@ -16,54 +16,32 @@
   <description>Hide stack trace from ordinary users for security-sensitive deployments</description>
 
   <properties>
-    <findbugs-maven-plugin.version>3.0.1</findbugs-maven-plugin.version>
     <findbugs.failOnError>true</findbugs.failOnError>
```
Comment (jglick) [inline pom.xml]: :ant: delete this, it is the default.

File: pom.xml [hunk 2]
```diff
@@ -16,54 +16,32 @@
...
   <dependencies>
     <dependency>
-      <groupId>com.google.code.findbugs</groupId>
-      <artifactId>annotations</artifactId>
-      <version>3.0.0</version>
-      <scope>provided</scope>
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>matrix-auth</artifactId>
+      <version>1.3.2</version>
+      <scope>test</scope>
```
Comment (jglick) [inline pom.xml]: :ant: take the opportunity to delete this dep and convert to `MockAuthorizationStrategy`.

File: src/test/java/org/jenkinsci/plugins/suppress_stack_trace/SuppressionFilterTest.java [hunk 3]
```diff
@@ -56,43 +49,44 @@ protected void setUp() throws Exception {
...
+        verifyLoginPage(wc.goTo("self/authenticationRequired"));
+        verifyLoginPage(wc.goTo("self/authenticationRequiredInJelly"));
     }
 
     private void verifyLoginPage(HtmlPage r) {
         String payload = r.getWebResponse().getContentAsString();
 
         assertEquals(payload, 200, r.getWebResponse().getStatusCode());
-        assertEquals(payload, "/login", r.getWebResponse().getUrl().getPath());
+        assertEquals(payload, "/jenkins/login", r.getUrl().getPath());
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/suppress_stack_trace/SuppressionFilterTest.java:82]: :ant: use

``` java
assertEquals(payload, r.contextPath + "/login", r.getUrl().getPath());
```

**Comment by jglick** [comment]
:bee:

**Comment by andresrc** [comment]
:bee: 

getting the `contextPath` fix proposed by Jesse in would be desirable.

**Comment by jglick** [comment]
re:bee:

**Comment by kwhetstone** [comment]
@reviewbybees done

**Comment by ghost** [comment]
This pull request has completed our internal processes and we now respectfully request the maintainers of this repository to consider our proposal contained within this pull request for merging.

**Comment by kwhetstone** [comment]
Is this ready to merge and cut a new release?

**Pull request merged**